File: c/resistor-color/resistor_color.c
#include "resistor_color.h"

uint16_t color_code(resistor_band_t band)
{
    return band;
}

const resistor_band_t *colors()
{
    static const resistor_band_t COLORS[] = {
        BLACK,
        BROWN,
        RED,
        ORANGE,
        YELLOW,
        GREEN,
        BLUE,
        VIOLET,
        GREY,
        WHITE,
    };

    return COLORS;
}

File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.Comparator;
import java.util.Map;
import java.util.TreeMap;

class ResistorColor {
    private static final Map<String, Integer> BANDS = new TreeMap<>() {{
        put("black", 0);
        put("brown", 1);
        put("red", 2);
        put("orange", 3);
        put("yellow", 4);
        put("green", 5);
        put("blue", 6);
        put("violet", 7);
        put("grey", 8);
        put("white", 9);
    }};

    int colorCode(String color) {
        if (!BANDS.containsKey(color)) {
            throw new IllegalArgumentException("unknown color.");
        }
        return BANDS.get(color);
    }

    String[] colors() {
        return BANDS.entrySet().stream()
                               .sorted(Comparator.comparingInt(Map.Entry::getValue))
                               .map(Map.Entry::getKey)
                               .toArray(String[]::new);
    }
}
